### PR TITLE
Create test for preventing infinite loop during processing of circulary dependent inline content items

### PR DESCRIPTION
**Pull request by daemoon**

### CODE REVIEW

File: KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            var retriever = new CodeFirstModelProvider(fakeDeliverClient);
+            retriever.TypeProvider = codeFirstTypeProvider;
+            A.CallTo(() => fakeDeliverClient.InlineContentItemsProcessor).Returns(processor);
+
+            var item = JToken.FromObject(rt1);
+            var modularContent = JToken.FromObject(modularContentObject);
+
+            var result = retriever.GetContentItemModel<ContentItemWithSingleRTE>(item, modularContent);
+
+            Assert.Equal("<span>FirstRT</span><span>SecondRT</span><span>FirstRT</span>", result.RT);
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs]: Is this really a correct behavior?
Shouldn't it be rather `<span>FirstRT</span><span>SecondRT</span>`?

File: KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        private class RichTextInlineResolver: IInlineContentItemsResolver<ContentItemWithSingleRTE>
+        {
+            public string Resolve(ResolvedContentItemData<ContentItemWithSingleRTE> data)
+            {
+                return data.Item.RT;
+            }
+        }
+
+        private static object rt1 = new
```
Comment (petrsvihlik) [inline KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs]: Could you add an extra test for an items that directly reference themselves? RT3->RT3

**Comment by petrsvihlik** [comment]
side note: internal reference = CMP-290

File: KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs (hunk 1 above)
Comment (daemoon) [inline KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs]: No, the first one in which circular dependency is processed as if there were no inline content items. This is first detected at rt1 where we find that we're processing object which is already being processed.

File: KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs (hunk 2 above)
Comment (daemoon) [inline KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs]: Yes

Comment (daemoon) [inline KenticoCloud.Delivery.Tests/CodeFirstModelProviderTests.cs]: Added

**Comment by petrsvihlik** [comment]
thx

**Pull request merged**